scene/2d/tile_map_layer.cpp

```diff
@@ -266,7 +266,11 @@ void TileMapLayer::_rendering_update(bool p_force_cleanup) {
 				// Sort the quadrant cells.
 				if (is_y_sort_enabled()) {
 					// For compatibility reasons, we use another comparator for Y-sorted layers.
-					rendering_quadrant->cells.sort_custom<CellDataYSortedComparator>();
+					if (x_draw_order_reversed) {
+						rendering_quadrant->cells.sort_custom<CellDataYSortedXReversedComparator>();
+					} else {
+						rendering_quadrant->cells.sort_custom<CellDataYSortedComparator>();
+					}
 				} else {
 					rendering_quadrant->cells.sort();
 				}
```

Comment:
As mentioned in https://github.com/godotengine/godot-proposals/issues/9882#issuecomment-2147664879 it could be something like:
```cpp
// Sort the quadrant cells.
if (is_y_sort_enabled() && x_draw_order_reversed) {
	rendering_quadrant->cells.sort_custom<CellDataYSortedXReversedComparator>();
} else {
	rendering_quadrant->cells.sort();
}
```
and `CellDataYSortedComparator` could be removed as it's the same as CellData's default comparator.

But this one is more explicit / more prone to changes. So fine too. Just mentioning in case you've missed it. :slightly_smiling_face: